app/Nova/User.php

```diff
@@ -809,4 +815,31 @@ private static function adminCanRun(NovaRequest $request): bool
     {
         return $request->user()->hasRole('admin');
     }
+
+    //Helper method: Graduation semester is a 6-digit code by default.
+    //Prepares semester data for display
+    private static function parseGradSemester(?string $sem): string
+    {
+        if ($sem === null) {
+            return '';
+        }
+        if (preg_match('/^[0-9]{4}0[258]$/', $sem) === false) {
```

Comment:
`preg_match` only returns `false` if there is an _error_. If the regex is valid but does not match the input string, it returns `0`. [[docs](https://www.php.net/manual/en/function.preg-match.php)]